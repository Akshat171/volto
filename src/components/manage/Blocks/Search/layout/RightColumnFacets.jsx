import React from 'react';
import { SearchInput, SearchDetails, Facets } from '../components';
import { Grid, Divider, Segment } from 'semantic-ui-react';
import { Button } from 'semantic-ui-react';
import { flushSync } from 'react-dom';

const FacetWrapper = ({ children }) => (
  <Segment basic className="facet">
    {children}
  </Segment>
);

const RightColumnFacets = (props) => {
  const {
    children,
    data,
    totalItems,
    facets,
    setFacets,
    onTriggerSearch,
    searchedText, // search text for previous search
    searchText, // search text currently being entered (controlled input)
    isEditMode,
    // searchData,
    // mode = 'view',
    // variation,
  } = props;
  const { showSearchButton } = data;
  const isLive = !showSearchButton;

  return (
    <Grid className="searchBlock-facets right-column-facets" stackable>
      <Grid.Row>
        <Grid.Column width={12}>
          <h3>{data.title}</h3>
          <SearchDetails text={searchedText} total={totalItems} />
        </Grid.Column>
      </Grid.Row>

      <Grid.Row columns="equal" verticalAlign="bottom">
        {data.showSearchInput && (
          <Grid.Column width={7}>
            <SearchInput {...props} isLive={isLive} />
          </Grid.Column>
        )}
        {data.showSearchButton && (
          <Grid.Column>
            <Button onClick={() => onTriggerSearch(searchText)}>
              {data.searchButtonLabel || 'Search!'}
            </Button>
          </Grid.Column>
        )}
      </Grid.Row>

      <Grid.Row>
        <Grid.Column mobile={12} tablet={8} computer={9}>
          <Divider />
          {children}
        </Grid.Column>

        <Grid.Column mobile={12} tablet={4} computer={3}>
          <Facets
            data={data}
            facets={facets}
            isEditMode={isEditMode}
            setFacets={(f) => {
              flushSync(() => {
                setFacets(f);
                if (isLive) onTriggerSearch(searchedText || '', f);
              });
            }}
            facetWrapper={FacetWrapper}
          />
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
};

export default RightColumnFacets;
