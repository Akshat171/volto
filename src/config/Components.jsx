import { PreviewImage } from '@plone/volto/components';
import TeaserDefaultBody from '@plone/volto/components/manage/Blocks/Teaser/DefaultBody';

export const components = {
  PreviewImage: { component: PreviewImage },
  'Teaser|Default': { component: TeaserDefaultBody },
};
