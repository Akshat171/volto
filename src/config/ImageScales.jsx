export const imageScales = {
  huge: 1600,
  great: 1200,
  larger: 1000,
  large: 800,
  teaser: 600,
  preview: 400,
  mini: 200,
  thumb: 128,
  tile: 64,
  icon: 32,
};
