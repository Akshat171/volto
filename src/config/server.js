import imagesMiddleware from '@plone/volto/express-middleware/images';
import filesMiddleware from '@plone/volto/express-middleware/files';
import robotstxtMiddleware from '@plone/volto/express-middleware/robotstxt';
import sitemapMiddleware from '@plone/volto/express-middleware/sitemap';
import devProxyMiddleware from '@plone/volto/express-middleware/devproxy';
import prefixPathMiddleware from '@plone/volto/express-middleware/prefixPath';

const settings = {
  expressMiddleware: [
    devProxyMiddleware(),
    prefixPathMiddleware(),
    filesMiddleware(),
    imagesMiddleware(),
    robotstxtMiddleware(),
    sitemapMiddleware(),
  ],
  criticalCssPath: 'public/critical.css',
  readCriticalCss: null, // so it will be defaultReadCriticalCss
  extractScripts: { errorPages: false },
};

export default settings;
