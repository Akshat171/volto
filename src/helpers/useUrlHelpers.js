import { useSelector } from 'react-redux';
import { _getBaseUrl } from './Url/Url';

export default function useUrlHelpers() {
  // pseudocode
  const headers = useSelector((state) => state.hostHeaders);
  const config = {
    settings: {
      example_apiPath: headers.host,
    },
  };

  return {
    getBaseUrl: _getBaseUrl(config),
  };
}
