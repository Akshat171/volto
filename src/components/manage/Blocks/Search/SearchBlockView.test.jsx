import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import config from '@plone/volto/registry';
import { getBaseUrl } from '@plone/volto/helpers';
import SearchBlockView from './SearchBlockView';
import RightColumnFacets from './layout/RightColumnFacets';

const subscriptionsblock_props = {
  variation: {
    id: 'facetsRightSide',
    title: 'Facets on right side',
    view: RightColumnFacets,
    isDefault: true,
  },
  id: 'abc123',
  data: {
    '@type': 'search',
    listingBodyTemplate: 'listing_for_admins',
    query: {
      b_size: '10',
      query: [
        {
          i: 'portal_type',
          o: 'plone.app.querystring.operation.selection.any',
          v: ['News Item'],
        },
      ],
      sort_on: 'effective',
      sort_order: 'descending',
      sort_order_boolean: true,
    },
    showSearchInput: true,
    showSortOn: true,
    sortOnLabel: 'Search for',
    sortOnOptions: ['modified'],
    variation: 'facetsRightSide',
    facets: [],
  },
  path: '/',
  querystring: {},
  facets: {},
};
subscriptionsblock_props.extensions =
  config.blocks.blocksConfig.listing.extensions;

beforeAll(() => {
  config.blocks.blocksConfig.listing = {
    ...config.blocks.blocksConfig.listing,
    variations: [
      ...config.blocks.blocksConfig.listing.variations,
      {
        id: 'listing_for_all',
        isDefault: false,
        title: 'Fancy listing variation for general use',
        template: () => <div className="mocked-general-listing-template"></div>,
      },
    ],
    variations_internal: [
      ...(config.blocks.blocksConfig.listing.variations_internal || []),
      {
        id: 'listing_for_admins',
        isDefault: false,
        title: 'Admin listing variation',
        template: () => <div className="mocked-admin-listing-template"></div>,
      },
    ],
  };
});

const mockStore = configureStore();

describe('SearchBlock', () => {
  it('renders a search block with special admin list variation template', () => {
    const store = mockStore({
      querystringsearch: {
        subrequests: {
          abc123: {
            items: [
              {
                '@id': 'http://localhost:8080/Plone/the-page',
                '@type': 'Document',
                description: '',
                review_state: 'private',
                title: 'the page',
              },
              {
                '@id': 'http://localhost:8080/Plone/front-page',
                '@type': 'Document',
                description:
                  'Congratulations! You have successfully installed Plone.',
                review_state: 'published',
                title: 'Welcome to Plone',
              },
            ],
            total: 2,
          },
        },
      },
      content: {
        data: {
          is_folderish: true,
        },
      },
      intl: {
        locale: 'en',
        messages: {},
      },
      breadcrumbs: [],
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <SearchBlockView
            {...subscriptionsblock_props}
            path={getBaseUrl(subscriptionsblock_props.path)}
            mode="view"
          />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
