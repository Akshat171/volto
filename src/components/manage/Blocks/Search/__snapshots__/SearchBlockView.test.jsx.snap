// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBlock renders a search block with special admin list variation template 1`] = `
<div
  className="block search"
>
  <div
    className="ui stackable grid searchBlock-facets right-column-facets"
  >
    <div
      className="row"
    >
      <div
        className="column"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="twelve wide computer twelve wide mobile twelve wide tablet column"
      >
        <div
          className="search-wrapper"
        >
          <div
            className="search-input"
          >
            <div
              className="ui fluid input"
            >
              <input
                id="abc123-searchtext"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Search"
                type="text"
                value=""
              />
            </div>
            <button
              className="ui basic icon button search-input-live-icon-button"
              onClick={[Function]}
            >
              <svg
                className="icon"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
                onClick={null}
                style={
                  Object {
                    "fill": "currentColor",
                    "height": "36px",
                    "width": "auto",
                  }
                }
                viewBox=""
                xmlns=""
              />
            </button>
          </div>
        </div>
        <div />
        <div
          className="search-results-count-sort"
        >
          <h4
            className="search-details"
          >
            
            Search results
            : 
            2
          </h4>
        </div>
        <div>
          <div
            className="mocked-admin-listing-template"
          />
        </div>
      </div>
      0
    </div>
  </div>
</div>
`;
